### PR TITLE
The PR that Makes this Module Useful

### PR DESCRIPTION
**Pull request by nelsonic**
Resolving Paths: https://github.com/dwyl/env2/issues/4
No Argument required: https://github.com/dwyl/env2/issues/5

### CODE REVIEW

**Comment by nelsonic** [comment]
@rjmk let us know what you think! 
```js
if(rafe === 'happy'){
rafe.merge();
}
```

File: lib/env.js [hunk 1]
```diff
@@ -1,8 +1,30 @@
...
-  try {
+/**
+ * load_environment_variables_from_file does exactly what its name suggests
+ * attempts to load a file containing a list of enviroment variables
+ * @param {String} filepath - the path to your env.json or .env file
+ * try/catch has a nice stack trace if we fail to find/load your the file
+ */
+module.exports = function load_environment_variables_from_file (filepath) {
+  try { // first we determine how to find the env.json or config.env file
+    var module_callee    = module.parent.id;
```
Comment (rjmk) [inline lib/env.js]: e.g. .../node_modules/env2/index.js (which requires this file)

File: lib/env.js [hunk 2]
```diff
@@ -1,8 +1,30 @@
...
+    var called_by_module = module_callee.indexOf('node_modules');
```
Comment (rjmk) [inline lib/env.js]: Root of project?

File: lib/env.js [hunk 3]
```diff
@@ -1,8 +1,30 @@
...
+    var parent_directory = module_callee.slice(0,called_by_module);
```
Comment (rjmk) [inline lib/env.js]: Root of project?

File: lib/env.js [hunk 4]
```diff
@@ -1,8 +1,30 @@
...
+    if(filepath) { // if required without a config.env file we search for one!
+      if(called_by_module > -1){
```
Comment (rjmk) [inline lib/env.js]: if substring 'node_modules' occurred

File: lib/env.js [hunk 5]
```diff
@@ -1,8 +1,30 @@
...
+        filepath = path.resolve(parent_directory + filepath);
```
Comment (rjmk) [inline lib/env.js]: api -- now reference path from root of project no matter where requiring from?

File: lib/env.js [hunk 6]
```diff
@@ -1,8 +1,30 @@
...
+      }
+    }
+    else { // https://github.com/dwyl/env2/issues/6
+      var dir = fs.readdirSync(parent_directory); // list of files in dir
+      dir.forEach(function(file) {  // search for a .env file in dir
+        if(file.indexOf('.env') > -1) { // found a .env file? use it!
+          filepath = parent_directory + file;
```
Comment (rjmk) [inline lib/env.js]: set default .env (no env.json?)

**Comment by rjmk** [comment]
AWESOME!

**Comment by nelsonic** [comment]
thanks @rjmk you've *made* it ***awesomer***!! :tada:

**Pull request merged**